### PR TITLE
Fix for Issue #128

### LINKED ISSUE #128: Bug when specifying port on the command line and running with GUI
I get the following error when specifying a port on the command line (with "-p <port>") if I have the GUI requirements installed. This is caused by "port" being parsed as a list of values rather than an integer (nargs="+" in the argument parser). I already have a patch to fix this, and will submit it shortly.

Traceback (most recent call last):
  File "SWProxy.py", line 265, in <module>
    win = gui.MainWindow(get_external_ip(), options.port)
  File "/Users/jake/Projects/SWProxy/SWParser/gui/gui.py", line 31, in __init__
    self.ui.proxyPort.setValue(port)
TypeError: QSpinBox.setValue(int): argument 1 has unexpected type 'list'
Segmentation fault: 11

### PR DESCRIPTION
**Pull request by jake-griffin**
fixes #128 - bug caused in gui mode when you specify a port

### CODE REVIEW

File: proxy.py [hunk 1]
```diff
@@ -534,11 +534,6 @@ class TCP(object):
     """TCP server implementation."""
 
     def __init__(self, hostname='127.0.0.1', port=8899, backlog=100):
-        # first clean port
-        if isinstance(port, list):
-            port = port[0]
-        if not isinstance(port, int):
-            port = int(port)
```
Comment (jake-griffin) [inline proxy.py]: This is no longer necessary now that port is always being passed in as an integer

**Comment by kakaroto** [comment]
I'm not a fan... first, why remove the help= argument ? and secondly, you should leave the list/int instance check and cast, but it shouldn't have been there, it should have been right after it parses the arguments. 
Also, wouldn't the "port" value possibly be a string, and that's why it needs to be cast into an int ?

**Comment by jake-griffin** [comment]
The 'help=' argument is pretty useless as it was written. What more does "Port Number" tell you on top of the argument name "port" itself? I can add it back in if you insist, but the other two arguments don't have any 'help=', so I was just being consistent. If you run "python SWProxy.py -h" with and without my change, you will see what this change actually does, and I believe it will make more sense after you see that.

As far as the list/string checks:
- Now that I removed 'nargs=', `port` will never be a list
- Because type=int, it will already be converted to an integer from a string
That is why I removed the check and cast. The only thing that it will fail on is if port is not an integer (like 'test'), and in that case it gives a meaningful error message at the time of parsing and doesn't even get to the check and cast code (SWProxy.py: error: argument -p/--port: invalid int value: 'test')

**Comment by jake-griffin** [comment]
Without my change:
```
[master] $ python SWProxy.py -h
usage: SWProxy.py [-h] [-d] [-g] [-p PORT]

SWParser

optional arguments:
  -h, --help            show this help message and exit
  -d, --debug
  -g, --no-gui
  -p PORT [PORT ...], --port PORT [PORT ...]
                        Port number
```
With my change:
```
[gui_port] $ python SWProxy.py -h
usage: SWProxy.py [-h] [-d] [-g] [-p PORT]

SWParser

optional arguments:
  -h, --help            show this help message and exit
  -d, --debug
  -g, --no-gui
  -p PORT, --port PORT
```

**Comment by kakaroto** [comment]
Ok, thanks for the explanation, seen that way, I guess it makes sense. I'm approving it.

**Pull request merged**